src/Components/common/contentEntitlement/index.js

```diff
@@ -0,0 +1,51 @@
+import i18n from "../../../i18n";
+import { usePelcro } from "../../../hooks/usePelcro";
+import { notify } from "../../../SubComponents/Notification";
+
+export const init = () => {
+  const { whenSiteReady } = usePelcro.getStore();
+
+  whenSiteReady(() => {
+    const entitlementsProtectedElements = document.querySelectorAll(
+      "[data-entitlement]"
+    );
+
+    entitlementsProtectedElements.forEach((elem) => {
+      const shouldBlurContent = !window.Pelcro.user.isEntitled(
+        elem.dataset.entitlement
+      );
+
+      if (shouldBlurContent) {
+        // remove all event listeners from the elem by replacing it with a deep clone of itself
+        const elemDeepClone = elem.cloneNode(true);
+        elem.replaceWith(elemDeepClone);
+
+        // disable keyboard interaction with blured content
+        elemDeepClone.addEventListener(
+          "keydown",
+          (event) => {
+            // allow tab key navigation to prevent focus trapping
+            const TAB_KEY_CODE = 9;
+            if (event.keyCode != TAB_KEY_CODE) {
+              event.returnValue = false;
+              return false;
+            }
+          },
+          true
+        );
+
+        elemDeepClone.setAttribute(
+          "style",
+          "filter:blur(3px) !important; pointer-events:none !important; user-select:none !important"
```

Comment:
No issues with browser support here?

Reply:
@ProgaPanda  nope, my overall approach was more complicated and it depended on `backdrop-filter` which is not widely supported, however, I changed it and I used a simpler one, the trade-off here is that my old approach would've covered more cases. I'm still uncomfortable about this implementation, that's why I mentioned that QA should extensively